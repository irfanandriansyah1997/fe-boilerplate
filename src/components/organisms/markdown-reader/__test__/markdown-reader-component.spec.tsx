import { render, waitFor } from '@testing-library/react';

import './markdown-reader-mock.spec';
import MarkdownReader from '..';

/**
 * MOCK_FETCH
 * @returns {Promise<Response>}
 */
const MOCK_FETCH = (input: RequestInfo): Promise<Response> =>
  new Promise((resolve) => {
    if (input === `sample-url`)
      resolve(({
        status: 200,
        text: () => `### Hello World \n Sample Text On Markdown`
      } as unknown) as Response);

    throw new Error(`Content Not Found`);
  });

describe(`Testing Markdown Reader`, () => {
  const tempFetch = global.fetch;

  beforeEach(() => {
    global.fetch = jest.fn().mockImplementation(MOCK_FETCH);
  });

  it(`Snapshot Testing`, async () => {
    const { getByTestId } = render(<MarkdownReader markdownUrl="sample-url" />);

    await waitFor(() => {
      expect(getByTestId(/markdown-code/)).toBeInTheDocument();
      expect(getByTestId(/markdown-code/)).toMatchSnapshot();
    });
  });

  afterEach(() => {
    global.fetch = tempFetch;
  });
});
