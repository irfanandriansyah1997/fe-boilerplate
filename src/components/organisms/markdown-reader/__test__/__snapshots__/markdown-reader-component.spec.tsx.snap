// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Markdown Reader Snapshot Testing 1`] = `
<div
  data-testid="markdown-code"
>
  <div>
    ### Hello World 
 Sample Text On Markdown
  </div>
</div>
`;
